webapp/components/survey/CategoryDetails/store/actions/useSetImportSummaryColumnDataType.js

```diff
@@ -0,0 +1,8 @@
+import { useCallback } from 'react'
+import { State } from '../state'
+
+export const useSetImportSummaryColumnDataType = ({ setState }) =>
+  useCallback(
+    ({ columnName, dataType }) => setState(State.assocImportSummaryColumnDataType({ columnName, dataType })),
```

Comment:
you are not passing the prev state. this might set the state to a function

Reply:
this is identical to this: setState(statePrev => State.assocImportSummaryColumnDataType({ columnName, dataType })(statePrev)) and it's more compact.
You cannot assign a function to the state.

Reply:
ops :P